### PR TITLE
added cypress files.

### PR DESCRIPTION
**Pull request by Naveen-Goud**
PR for the cypress 

[BC98-79](https://zemoso.atlassian.net/browse/BC98-79)

### CODE REVIEW

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+const sellPageHands=new SellPageHands();
+const buyPageEyes=new BuyPageEyes();
+const buyPageHands=new BuyPageHands();
+ 
+ describe('minet tesing', () => {  
+    it("test from login page",()=>{
+      loginPage.visitLoginPage()
+      loginEyes.AssertTitle()
+      loginHands.EnterEmail("email","shaikshakeelgr@gmail.com")
+      loginHands.EnterPassword("password","shaik123@")
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: make the actions chainable for login hands. read the values from env file. avoid hard coding

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      loginHands.ClickOnSignup()
+       cy.wait(2000)
+ 
+      dashboardPageBaseEyes.AssertTitle()
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: make sure the actions chainable. check across the code

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      dashboardPageBaseEyes.AssertTitles()
+      dashboardPageBaseEyes.AssertButtons
+      dashboardPageBaseEyes.AssertTitles()
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: Maintain different it blocks for different scenaios to easily understand the code

File: cypress_bc98/cypress/robots/BaseRobot.ts [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    return this;
+  }
+
+  accessUrl(url: string) {
+    cy.visit(`${url}`);
+    return this;
+  }
+
+  login() {
+    cy.get('#email').type(Cypress.env('USER_NAME'));
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/robots/BaseRobot.ts:233]: Read values from where the function called

File: cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+     DataChangeAfterSelecting(){
+        baseRobotEyes.getByText("Ethereum wallet")
+         
+    }
+    AssertCheckoutPage(){
+        baseRobotEyes.getByText("Purchase is completed, please check your balance in your crypto wallet")
+        baseRobotEyes.getByText("BUY CRYPTO")
+    }
+    AssertWatchData(){
+        baseRobotEyes.getByText("USD Coin")
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts]: iterate the loop for common action with different values

File: cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { BaseEyes, BaseHands } from "../BaseRobot"; 
+import { Dependencies, RobotEyes, RobotHands } from "./Robot";
+
+const baseRobotEyes = new RobotEyes();
+const baseRobotHands = new RobotHands();
+
+
+export class LoginPageDepandence extends Dependencies{
+       visitLoginPage(){
+        this.accessUrl("https://bc98fe.bootcamp64.tk/login" );
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts]: avoid hard coding of urls

File: cypress_bc98/cypress/robots/minetApp/Sellpage.cy.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+
+export class SellPageEyes extends BaseEyes{
+    
+    AssertTitle(){
+        baseRobotEyes.getByText("Checkout")
+    }
+    
+    AssertElements(){
+        baseRobotEyes.getByText("Bitcoin")
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/robots/minetApp/Sellpage.cy.ts:15]: iterate loop for common action with different values

File: cypress_bc98/cypress/robots/minetApp/Sellpage.cy.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+         
+    }
+
+    AssertCheckoutPage(){
+        baseRobotEyes.getByText("Sell is completed, please check your balance in your Rupee coin")
+        baseRobotEyes.getByText("SELL CRYPTO")
+    }
+
+    AssertWatchData(){
+        baseRobotEyes.getByText("USD Coin")
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/robots/minetApp/Sellpage.cy.ts:33]: check this as well

File: cypress_bc98/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,109 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
```
Comment (srilakshmi-gadde) [inline cypress_bc98/tsconfig.json]: remove unnecessary commented lines

File: cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts (hunk 6 above)
Comment (Naveen-Goud) [reply cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts]: resolved

File: cypress_bc98/cypress/robots/BaseRobot.ts (hunk 4 above)
Comment (Naveen-Goud) [reply cypress_bc98/cypress/robots/BaseRobot.ts:233]: this was not used and present in the baserobot by default.

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts (hunk 1 above)
Comment (Naveen-Goud) [reply cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: values used from the env

File: cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts (hunk 5 above)
Comment (Naveen-Goud) [reply cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts]: can use iterate but testing many components and functions are made according the page flow.

File: cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts (hunk 6 above)
Comment (Naveen-Goud) [reply cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts]: declared in env file and imported it.

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts (hunk 1 above)
Comment (srilakshmi-gadde) [reply cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: Can you remove the hard coded values of email & password.

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      loginPage.visitLoginPage()
+      loginEyes.AssertTitle()
+      loginHands.EnterEmail("email","shaikshakeelgr@gmail.com")
+      loginHands.EnterPassword("password","shaik123@")
+      loginHands.ClickOnSignup()
+       cy.wait(2000)
+ 
+      dashboardPageBaseEyes.AssertTitle()
+      dashboardPageBaseEyes.AssertTitles()
+      dashboardPageBaseEyes.AssertButtons
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: Here "AssertButtons" is it a method. Have you run this code from your local?

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts (hunk 2 above)
Comment (srilakshmi-gadde) [reply cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: Make the actions chainable like this
 dashboardPageBaseEyes.AssertTitles()
      .AssertButtons()

File: cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+      dashboardPageBaseEyes.AssertTitles()
+ 
+      allAssetsEyes.AssertHeaders()
```
Comment (srilakshmi-gadde) [inline cypress_bc98/cypress/e2e/minet/loginRobots.cy.ts]: Make sure to maintain different it blocks for each scenario

File: cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts (hunk 6 above)
Comment (srilakshmi-gadde) [reply cypress_bc98/cypress/robots/minetApp/LoginPage.cy.ts]: I'm still seeing the url hardcoded.

File: cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts (hunk 5 above)
Comment (srilakshmi-gadde) [reply cypress_bc98/cypress/robots/minetApp/BuyPage.cy.ts]: if the same action used for different values, we can iterate the loop for the action/ method with different values using array

**Review by srilakshmi-gadde — APPROVED** [review]

**Pull request merged**